src/main/java/org/spongepowered/api/resource/ResourceManager.java

```diff
@@ -48,17 +52,26 @@
      * Gets a loaded resource at the given path, or {@link Optional#empty()}
      * if it does not exist.
      *
+     * <p>In the pack, the path will point to a resource. The resource should
+     * be located roughly at {@code data/namespace/path}</p>
+     *
      * @param path The path to the resource
      * @return The resource
+     * @throws IOException If there was an error reading the resource
+     * @throws FileNotFoundException If the resource does not exist.
      */
-    Resource getResource(ResourcePath path) throws IOException;
+    Resource getResource(ResourcePath path) throws IOException, FileNotFoundException;
 
     /**
-     * Returns all of the resources which exist in each {@link Pack}.
+     * Returns all of the resources which exist in each {@link Pack}. Remember
+     * to close all of the resources.
      *
      * @return All of the resources
+     * @see #getResource(ResourcePath)
+     * @throws IOException If there was an error reading any of the resources
+     * @throws FileNotFoundException If no resources exist
      */
-    List<Resource> getResources(ResourcePath path);
+    List<Resource> getResources(ResourcePath path) throws IOException, FileNotFoundException;
```

Comment:
Remember to document the order of the list is based on priority. Higher priority entries are placed earlier in this list (I think this is the case; I may be wrong).